core/src/main/java/org/kiwix/kiwixmobile/core/main/CompatFindActionModeCallback.kt

```diff
@@ -0,0 +1,213 @@
+/*
+ * Kiwix Android
+ * Copyright (c) 2019 Kiwix <android.kiwix.org>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+package org.kiwix.kiwixmobile.core.main
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.text.Editable
+import android.text.Selection
+import android.text.Spannable
+import android.text.TextWatcher
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.Menu
+import android.view.MenuItem
+import android.view.View
+import android.view.inputmethod.InputMethodManager
+import android.webkit.WebView
+import android.widget.EditText
+import android.widget.TextView
+import androidx.appcompat.view.ActionMode
+import org.kiwix.kiwixmobile.core.R
+
+class CompatFindActionModeCallback internal constructor(context: Context) :
+  ActionMode.Callback, TextWatcher, View.OnClickListener {
+  private val tag = "CompatFindActionMode"
+  @JvmField var isActive: Boolean
+
+  @SuppressLint("InflateParams")
+  private val customView: View = LayoutInflater.from(context).inflate(R.layout.webview_search, null)
+  private val editText: EditText
+  private val findResultsTextView: TextView
+  private var webView: WebView? = null
+  private val input: InputMethodManager
+  private var actionMode: ActionMode? = null
+
+  init {
+    editText = customView.findViewById(R.id.edit)
```

Comment:
we can move these field initalisations to the fields themselves instead of in the init block

Reply:
8fa8f40